x-pack/plugins/apm/public/components/routing/templates/apm_main_template.tsx

```diff
@@ -54,16 +54,26 @@ export function ApmMainTemplate({
     return callApmApi('GET /internal/apm/has_data');
   }, []);
 
+  const shouldBypassNoDataScreen = bypassNoDataScreenPaths.some((path) =>
+    location.pathname.includes(path)
+  );
+
+  const { data: fleetApmIntegrations } = useFetcher(
+    (callApmApi) => {
+      if (!data?.hasData && !shouldBypassNoDataScreen) {
+        return callApmApi('GET /internal/apm/fleet/has_data');
+      }
+    },
+    [shouldBypassNoDataScreen, data?.hasData]
+  );
+
   const noDataConfig = getNoDataConfig({
     basePath,
     docsLink: docLinks!.links.observability.guide,
     hasData: data?.hasData,
+    hasApmIntegrations: fleetApmIntegrations?.hasData,
```

Comment:
```suggestion
    hasApmPolicies: fleetApmIntegrations?.has_apm_policies,
```

Reply:
Actualy, while you are at it, you should also add `shouldBypassNoDataScreen` to `getNoDataConfig`. This way we can keep all the logic co-located

```suggestion
    hasApmPolicies: fleetApmIntegrations?.hasApmPolicies,
    shouldBypassNoDataScreen
```


```ts
export function getNoDataConfig({
  docsLink,
  basePath,
  hasApmData,
  shouldBypassNoDataScreen,
}: {
  docsLink: string;
  basePath?: string;
  hasApmData?: boolean;
  shouldBypassNoDataScreen: boolean;
}): KibanaPageTemplateProps['noDataConfig'] {
  // don't show "no data screen" when there is APM data or it should be bypassed
  if (hasApmData || shouldBypassNoDataScreen) {
    return;
  }

  return {
   // ...
  };
}

```

Reply:
Changed this in https://github.com/elastic/kibana/pull/126146/commits/6d18c704e209c04155d0430b197d3c958741154d